contrib/scripts/setup.sh

```diff
@@ -0,0 +1,188 @@
+      user_sh=${user_sh:-none}
+      user_sh=$(basename $user_sh)
+   fi
+else
+  user_sh=$(basename $SHELL)
+fi
+
+bash_exist=1
+case $user_sh in
+	zsh)
+		sh_profile=$HOME/.zshrc
+		;;
+	bash)
+		sh_profile=$HOME/.bash_profile
+		;;
+	sh)
+		sh_profile=$HOME/.profile
+		;;
+	csh | tcsh)
+		sh_profile=$HOME/.cshrc
+		;;
+	*)
+		# if $sh_profile is null install script will stop following grep
+		echo "$user_sh: unknown SHELL, force set to /bin/bash"
+		sh_profile=$HOME/.bash_profile
+		CUBRID_SH_INSTALLED=0
+		;;
+esac
+
+if [ ! -f $sh_profile ];then
+  bash_exist=0
+  touch $sh_profile
+fi
+
+append_profile=$(grep "${PRODUCT_NAME} environment" $sh_profile)
+
+if [ -z "$append_profile" ];then
+  echo '#-------------------------------------------------------------------------------' >> $sh_profile
+  if [ $? -ne 0 ];then
+    CUBRID_SH_INSTALLED=0
+    echo "Please check your permission for file $sh_profile"
+  else
+    echo '# set '${PRODUCT_NAME}' environment variables'                                    >> $sh_profile
+    echo '#-------------------------------------------------------------------------------' >> $sh_profile
+
+    case $user_sh in
+      bash | sh)
+        echo 'if [ -f $HOME/.cubrid.sh ];then'                                              >> $sh_profile
+        echo '. $HOME/.cubrid.sh'                                                           >> $sh_profile
+        echo 'fi'                                                                           >> $sh_profile
+        ;;
+      csh | tcsh)
+        echo 'if ( -f ~/.cubrid.csh ) then' 		>> $sh_profile
+        echo '  source ~/.cubrid.csh' 			>> $sh_profile
+        echo 'endif' 					>> $sh_profile
+        ;;
+      zsh)
+        echo 'if [ -f $HOME/.cubrid.sh ];then'                                              >> $sh_profile
+        echo '  source $HOME/.cubrid.sh'                                                    >> $sh_profile
+        echo 'fi'                                                                           >> $sh_profile
+        ;;
+      *)
+        CUBRID_SH_INSTALLED=0
+        ;;
+    esac
+  fi	# $? - ne 0
+fi	# -z "$append_profile"
+
+if [ $CUBRID_SH_INSTALLED -eq 1 ] && [ $bash_exist -eq 0 ];then
+  echo "Notification: $sh_profile is created"
+fi
+
+# create a demo db
+echo ""
+if [ -x "${cubrid_home}/demo/make_cubrid_demo.sh" ]; then
+  . ${cubrid_sh_envfile}
+  (mkdir -p $CUBRID_DATABASES/demodb && cd $CUBRID_DATABASES/demodb && $CUBRID/demo/make_cubrid_demo.sh > /dev/null 2>&1)
+  if [ $? = 0 ]; then
+    echo "demodb has been successfully created."
+  else
+    echo "Cannot create demodb."
+  fi
+else
+  echo "${cubrid_home}/demo/make_cubrid_demo.sh : No such file"
+fi
+
+echo ""
+echo "If you want to use CUBRID, run the following command to set required environment variables."
+if [ $CUBRID_SH_INSTALLED -eq 0 ];then
+        echo "(or you can add the command into your current shell profile file to set permanently)"
+fi
```

Comment:
If CUBRID_SH_INSTALLED is 0, I think that it is better to exit after printing an error message.

Reply:
Ok, I will change it.